src/convenience_api.rs

```diff
@@ -0,0 +1,275 @@
+use ffi::{MediaInfo, MediaInfoResult, MediaInfoStream};
+use streams::{GeneralStream, VideoStream, AudioStream, ImageStream, MenuStream, OtherStream, TextStream};
+
+use chrono::{UTC, DateTime};
+use std::rc::Rc;
+use std::cell::RefCell;
+use std::path::Path;
+use std::time::Duration;
+
+pub struct MediaInfoWrapper {
+    general_stream: GeneralStream,
+    handle: Rc<RefCell<MediaInfo>>,
+    video_streams: Option<Vec<VideoStream>>,
+    audio_streams: Option<Vec<AudioStream>>,
+    image_streams: Option<Vec<ImageStream>>,
+    menu_streams: Option<Vec<MenuStream>>,
+    text_streams: Option<Vec<TextStream>>,
+    other_streams: Option<Vec<OtherStream>>,
+}
+
+impl MediaInfoWrapper {
+    pub fn new() -> MediaInfoWrapper {
+        MediaInfoWrapper {
+            general_stream: GeneralStream {
+                stream_type: MediaInfoStream::General,
+                handler: None,
+            },
+            video_streams: None,
+            audio_streams: None,
+            image_streams: None,
+            menu_streams: None,
+            text_streams: None,
+            other_streams: None,
+            handle: Rc::new(RefCell::new(MediaInfo::new())),
+        }
+    }
+
+    pub fn open(&mut self, path: &Path) -> MediaInfoResult<usize> {
+        let result = self.handle.borrow_mut().open(path);
+
+        match result {
+            Ok(r) => {
+                self.wrap_streams();
+                Ok(r)
+            },
+            Err(r) => Err(r),
+        }
+    }
+
+    pub fn from_data(&mut self, data: &[u8]) -> Result<(), String>{
```

Comment:
Not sure that I understand this function, `self` is passed in but then a new `MediaInfo` handle is created. Shouldn't it be using `self.handle` instead of `minfo` below? In my experience `from_*` functions do not take `self` but construct `Self` from the data that is passed in.

Reply:
This method is supposed to take the data from `data: &u[8]` instead of from the filesystem. 

What you found is left overs from changing the API a couple of times. Maybe I should a a test for this method as well.